android/java/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java

```diff
@@ -207,6 +208,62 @@ public void setGroup(String name, String[] value, EventOptions options, Middlewa
         this.client.setGroup(name, jsonValue, extra);
     }
 
+    public void groupIdentify(String groupType, String groupName, Group event) {
+        this.groupIdentify(groupType, groupName, event, null, null);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, EventOptions options) {
+        this.groupIdentify(groupType, groupName, event, options, null);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, MiddlewareExtra extra) {
+        this.groupIdentify(groupType, groupName, event, null, extra);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, EventOptions options, MiddlewareExtra extra) {
+        if (!this.isInitializedAndEnabled()) {
+            return;
+        }
+        this.handleEventOptions(options, null);
+        com.amplitude.api.Identify identify = new com.amplitude.api.Identify();
+        JSONObject groupProperties = this.getEventPropertiesJson(event);
+        if (groupProperties != null) {
+            Iterator<String> keys = groupProperties.keys();
+            while (keys.hasNext()) {
+                String key = keys.next();
+
+                Object value;
+                try {
+                    value = groupProperties.get(key);
+                } catch (JSONException e) {
+                    System.err.println("Can't get value for a group property '" + key + "': " + e);
+                    continue;
+                }
+
+                if (value instanceof Boolean) {
```

Comment:
Can we use AmplitudeClient.truncate() to avoid all these type checks?

Reply:
@falconandy @qingzhuozhen if this works this might be a better solution than the new `groupIdentify(groupType, grouptName, JSONObject groupProperties)` as it would be backwards compatiable with old versions of the Amplitude-Android.

Reply:
@justin-fiedler The `truncate()` method truncates string properties (including nested) and leave other properties as-is.

Reply:
@justin-fiedler @qingzhuozhen 
`groupIdentify(groupType, grouptName, JSONObject groupProperties)` is added
https://github.com/amplitude/Amplitude-Android/pull/324